spec/requests/api/editor/editor_can_put_articles_to_archive_spec.rb

```diff
@@ -1,37 +1,41 @@
-RSpec.describe 'DELETE /api/articles/:id' do
+RSpec.describe 'PUT /api/articles/:id' do
   let(:article) { create(:article) }
   let(:editor) { create(:user, role: 'editor') }
   let(:editor_headers) { editor.create_new_auth_token }
 
   describe 'Successfully' do
     before do
-      delete "/api/articles/#{article.id}", headers: editor_headers
+      put "/api/articles/#{article.id}",
+          params: { status: 'archived' },
+          headers: editor_headers
     end
 
     it 'is expected to return status code 200' do
       expect(response).to have_http_status 200
     end
 
     it 'is expected to return a success message' do
-      expect(response_json['message']).to eq 'The article was successfully deleted'
+      expect(response_json['message']).to eq 'Your article has been successfully updated!'
```

Comment:
Ain't it better to set the message to be more informative? like `The article has been successfully archived!`

Reply:
true, but it will require some new conditions for that, as now deleting is in the same PUT request as publishing so we need to differentiate between them. We thought that it aint worth it.

Reply:
Agree :3